pkg/services/repo_service.go

```diff
@@ -66,3 +72,76 @@ func (a *argoCDService) GetApps(ctx context.Context, repoURL string, revision st
 
 	return res, nil
 }
+
+func (a *argoCDService) checkoutRepo(gitClient git.Client, revision string) error {
+	err := gitClient.Init()
+	if err != nil {
+		return errors.Wrap(err, "Error during initiliazing repo")
+	}
+
+	err = gitClient.Fetch()
+	if err != nil {
+		return errors.Wrap(err, "Error during fetching repo")
+	}
+
+	commitSHA, err := gitClient.LsRemote(revision)
+	if err != nil {
+		return errors.Wrap(err, "Error during fetching commitSHA")
+	}
+	err = gitClient.Checkout(commitSHA)
+	if err != nil {
+		return errors.Wrap(err, "Error during repo checkout")
+	}
+	return nil
+}
+
+func (a *argoCDService) GetPaths(ctx context.Context, repoURL string, revision string, pattern string) ([]string, error) {
```

Comment:
Can we use GetPathes to get the applications? Does get path "*" will return all the top level directories and files?

Reply:
`GetPaths` uses `git ls-files` under the hood, and it returns all the paths in the directory tree, matching the specified pattern, and not just the top level files or directories. So, it will need some extra work to retrieve applications using `ls-files`. We will also need to check if a directory is a valid `app` directory, by checking if it contains `helm` or `kustomize` artifacts. Since, ArgoCD repo server already returns the `apps` present in a repo, I think we should just piggy back on that functionality.

Reply:
An application can be plain k8s objects (without Helm / Kustomize). Currently, the repo server will not return such directories, that's was the reason I asked. We should think about it after you merge this code.